lib/HLSL/DxilValidation.cpp

```diff
@@ -2134,6 +2134,24 @@ static void ValidateBarrierFlagArg(ValidationContext &ValCtx, CallInst *CI,
   }
 }
 
+std::string GetLaunchTypeStr(DXIL::NodeLaunchType LT) {
+  switch (LT) {
+	case DXIL::NodeLaunchType::Broadcasting:
+		return "Broadcasting";
+  case DXIL::NodeLaunchType::Coalescing:
+		return "Coalescing";
+  case DXIL::NodeLaunchType::Thread:
+    return "Thread";
+  case DXIL::NodeLaunchType::Invalid:
+    return "Invalid";
+  case DXIL::NodeLaunchType::LastEntry:
+    return "LastEntry";
+	
+	default:
+		return "Unknown";
+	}
```

Comment:
I believe all of these could be considered `"<invalid>"` which should never be seen unless some other part of validation is not performed.  Also, there's weird line spacing and tabbed indentation here.